File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/java/src/lojaexemplo/Loja.java
package lojaexemplo;

import java.util.ArrayList;

public class Loja {

    private ArrayList<Produto> produtos;

    public Loja() {
        produtos = new ArrayList<>();
    }

    public void adicionarProduto(Produto p) {
        produtos.add(p);
        System.out.println("Produto adicionado com sucesso!");
    }

    public void listarProdutos() {
        if (produtos.isEmpty()) {
            System.out.println("Nenhum produto cadastrado.");
            return;
        }

        for (Produto p : produtos) {
            System.out.println(p);
        }
    }

    public Produto buscarProduto(int codigo) {
        for (Produto p : produtos) {
            if (p.getCodigo() == codigo) {
                return p;
            }
        }
        return null;
    }

    public void removerProduto(int codigo) {
        Produto p = buscarProduto(codigo);
        if (p != null) {
            produtos.remove(p);
            System.out.println("Produto removido.");
        } 
        
        else {
            System.out.println("Produto não encontrado.");
        }
    }

    public double calcularValorTotalEstoque() {
        double total = 0;
        for (Produto p : produtos) {
            total += p.calcularTotal();
        }
        return total;
    }

    public Produto produtoMaisCaro() {
        if (produtos.isEmpty()) return null;

        Produto maisCaro = produtos.get(0);

        for (Produto p : produtos) {
            if (p.getPreco() > maisCaro.getPreco()) {
                maisCaro = p;
            }
        }
        return maisCaro;
    }
}

File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/csharp/LojaExemplo/Loja.cs
using System;
using System.Collections.Generic;

namespace LojaExemplo
{
    public class Loja
    {
        private List<Produto> produtos;

        public Loja()
        {
            produtos = new List<Produto>();
        }

        public void AdicionarProduto(Produto p)
        {
            produtos.Add(p);
            Console.WriteLine("Produto adicionado com sucesso!");
        }

        public void ListarProdutos()
        {
            if (produtos.Count == 0)
            {
                Console.WriteLine("Nenhum produto cadastrado.");
                return;
            }

            foreach (var p in produtos)
            {
                Console.WriteLine(p);
            }
        }

        public Produto BuscarProduto(int codigo)
        {
            return produtos.Find(p => p.Codigo == codigo);
        }

        public void RemoverProduto(int codigo)
        {
            Produto p = BuscarProduto(codigo);
            if (p != null)
            {
                produtos.Remove(p);
                Console.WriteLine("Produto removido.");
            }
            else
            {
                Console.WriteLine("Produto não encontrado.");
            }
        }

        public double CalcularValorTotalEstoque()
        {
            double total = 0;
            foreach (var p in produtos)
            {
                total += p.CalcularTotal();
            }
            return total;
        }

        public Produto ProdutoMaisCaro()
        {
            if (produtos.Count == 0) return null;

            Produto maisCaro = produtos[0];
            foreach (var p in produtos)
            {
                if (p.Preco > maisCaro.Preco)
                {
                    maisCaro = p;
                }
            }
            return maisCaro;
        }
    }
}
``